### PR TITLE
added parslet parse rules for mathml string

### LINKED ISSUE #14: Add MathML parser
Add parser for parsing **```MathML```** string to the ```syntax tree```.

### PR DESCRIPTION
**Pull request by suleman-uzair**
Added **```Parslet```** parse rules for parsing **```MathMl```** string

closes #14

### CODE REVIEW

File: lib/plurimath/mathml.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def initialize(text)
     end
 
     def to_formula
-      # TODO: Will be implemented soon
+      Plurimath::Mathml::Parser.new(text).parse
```
Comment (HassanAkbar) [inline lib/plurimath/mathml.rb:12]: Will this work currently?

File: lib/plurimath/mathml/parse.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          expr | str(tag)
+        end
+      end
+
+      def parse_symbols
+        Constants::SYMBOLS.keys.reduce do |expr, tag|
+          expr = str(expr) if expr.is_a?(Symbol)
+          expr | str(tag)
+        end
+      end
```
Comment (HassanAkbar) [inline lib/plurimath/mathml/parse.rb:45]: These functions and the `tags` function on line 65 have similar code can we merge these into a common method?

**Review by HassanAkbar — CHANGES_REQUESTED** [review]
Just some minor comments. Other than that it looks good. 👍

File: lib/plurimath/mathml/parse.rb (hunk 2 above)
Comment (ronaldtse) [reply lib/plurimath/mathml/parse.rb:45]: Agree

File: spec/plurimath/mathml/parse_spec.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                <mi>x</mi>
+                <mi>s</mi>
+              </munderover>
+            </mrow>
+          </mstyle>
+        </math>
+      MATHML
+    }
+    it "returns formula of sum and prod" do
+      expected_value = {
```
Comment (ronaldtse) [inline spec/plurimath/mathml/parse_spec.rb:134]: Is it possible to make this comparison more simple? For example, if the structure changes in the future, these compares will be very tough to update. You should be testing for particular outcomes, not data equivalence of the structures.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: spec/plurimath/mathml/parse_spec.rb (hunk 3 above)
Comment (ronaldtse) [reply spec/plurimath/mathml/parse_spec.rb:134]: But we do do that in a future PR, too.

File: spec/plurimath/mathml/parse_spec.rb [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    let (:exp) { "<math xmlns='http://www.w3.org/1998/Math/MathML'></math>" }
+
+    it "matches open and close tag" do
+      expect(formula[:open]).to eq("math")
+      expect(formula[:close]).to eq("math")
+    end
+
+    it "matches attributes" do
+      expect(formula[:attributes].first[:name]).to eq("xmlns")
+      expect(formula[:attributes].last[:value]).to eq("http://www.w3.org/1998/Math/MathML")
```
Comment (HassanAkbar) [inline spec/plurimath/mathml/parse_spec.rb:19]: I think both should be consistent. Currently, it feels like the name and value are for different attributes.

```suggestion
      expect(formula[:attributes].first[:name]).to eq("xmlns")
      expect(formula[:attributes].first[:value]).to eq("http://www.w3.org/1998/Math/MathML")
```

**Review by HassanAkbar — APPROVED** [review]
LGTM. 👍

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thanks @suleman-uzair !

**Pull request merged**